app/src/androidTest/java/com/duckduckgo/app/notification/NotificationSchedulerTest.kt

```diff
@@ -52,24 +55,47 @@ class NotificationSchedulerTest {
     }
 
     @Test
-    fun whenClearNotificationCanShowThenNotificationScheduled() = runBlocking<Unit> {
+    fun whenBothPrivacyNotificationAndCleatDataCanShowThenPrivacyNotificationScheduled() = runBlocking<Unit> {
+        whenever(privacyNotification.canShow()).thenReturn(true)
         whenever(clearNotification.canShow()).thenReturn(true)
         testee.scheduleNextNotification()
-        assertTrue(notificationScheduled())
+        assertTrue(notificationScheduled(PrivacyNotificationWorker::class.jvmName))
     }
 
     @Test
-    fun whenClearNotificationCannotShowThenNotificationNotScheduled() = runBlocking<Unit> {
+    fun whenPrivacyNotificationCanShowAndCleatDataCannotThenPrivacyNotificationScheduled() = runBlocking<Unit> {
+        whenever(privacyNotification.canShow()).thenReturn(true)
+        whenever(clearNotification.canShow()).thenReturn(false)
+        testee.scheduleNextNotification()
+        assertTrue(notificationScheduled(PrivacyNotificationWorker::class.jvmName))
+    }
+
+    @Test
+    fun whenPrivacyNotificationCannotShowAndClearNotificationCanShowThenNotificationScheduled() = runBlocking<Unit> {
+        whenever(privacyNotification.canShow()).thenReturn(false)
+        whenever(clearNotification.canShow()).thenReturn(true)
+        testee.scheduleNextNotification()
+        assertTrue(notificationScheduled(ClearDataNotificationWorker::class.jvmName))
+    }
+
+    @Test
+    fun whenNoNotificationCanShowThenNoNotificationScheduled() = runBlocking<Unit> {
+        whenever(privacyNotification.canShow()).thenReturn(false)
         whenever(clearNotification.canShow()).thenReturn(false)
         testee.scheduleNextNotification()
         assertFalse(notificationScheduled())
     }
 
-    private fun notificationScheduled(): Boolean {
-        return WorkManager
+    private fun notificationScheduled(workerName: String? = null): Boolean {
+        val workers = WorkManager
             .getInstance()
             .getWorkInfosByTag(NotificationScheduler.WORK_REQUEST_TAG)
             .get()
-            .any { it.state == WorkInfo.State.ENQUEUED }
+            .filter { it.state == WorkInfo.State.ENQUEUED }
+
+        if (workerName == null) {
```

Comment:
can you explain the changes made in here?

Reply:
We originally only had one type of notification, now that we have a second we need to check that the correct one has been queued, it's a bit clunky. I can rework it if you think it's unreadable.

Reply:
Ah, got you. yeah, something making it tricky for me to scan and understand it quickly. i wrote a suggestion above that would help my understanding of the code... lmk if that works for you